File: searchIndex-platform/module/search-api/search-manager/app/controllers/LoadDefinitionCacheController.java
package controllers;

import org.ekstep.common.dto.Request;

import managers.PlaySearchManager;
import play.mvc.Result;

public class LoadDefinitionCacheController extends SearchBaseController {
	
	private PlaySearchManager mgr = new PlaySearchManager();

	public Result loadDefinitionCache() {
		String apiId = "composite-search.loadDefinitionCache";
		Request request = getRequest(request().body(), apiId, request().uri());
		String response = mgr.callResyncDefinition(request);
		return ok(response).as("application/json");
	}

}

File: searchIndex-platform/module/search-api/search-manager/app/controllers/LoadDefinitionCacheController.scala
package controllers

import akka.actor.{ActorRef, ActorSystem}
import javax.inject.{Inject, Named}
import play.api.mvc.ControllerComponents

import scala.concurrent.{ExecutionContext, Future}
import managers.PlaySearchManager

class LoadDefinitionCacheController@Inject()(@Named("SearchManager") actor: ActorRef, cc: ControllerComponents, actorSystem: ActorSystem)(implicit exec: ExecutionContext) extends BaseController(cc)  {

    private val mgr = new PlaySearchManager

    def loadDefinitionCache() = Action.async { implicit request =>
        val apiId = "composite-search.loadDefinitionCache"
        val internalReq = getRequest(apiId)
        val response = mgr.callResyncDefinition(internalReq, actor)
        getResult(Future(response))
    }
}
